src/ImageSharp/Formats/Webp/Lossless/Vp8LHistogram.cs

```diff
@@ -1,63 +1,71 @@
+    private const int BlueSize = WebpConstants.NumLiteralCodes;
+    private const int AlphaSize = WebpConstants.NumLiteralCodes;
+    private const int DistanceSize = WebpConstants.NumDistanceCodes;
+    public const int LiteralSize = WebpConstants.NumLiteralCodes + WebpConstants.NumLengthCodes + (1 << WebpConstants.MaxColorCacheBits) + 1;
+    private const int UsedSize = 5; // 5 for literal, red, blue, alpha, distance
+    public const int BufferSize = RedSize + BlueSize + AlphaSize + DistanceSize + LiteralSize + UsedSize;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="Vp8LHistogram"/> class.
     /// </summary>
-    /// <param name="other">The histogram to create an instance from.</param>
-    private Vp8LHistogram(Vp8LHistogram other)
-        : this(other.PaletteCodeBits)
-    {
-        other.Red.AsSpan().CopyTo(this.Red);
-        other.Blue.AsSpan().CopyTo(this.Blue);
-        other.Alpha.AsSpan().CopyTo(this.Alpha);
-        other.Literal.AsSpan().CopyTo(this.Literal);
-        other.Distance.AsSpan().CopyTo(this.Distance);
-        other.IsUsed.AsSpan().CopyTo(this.IsUsed);
-        this.LiteralCost = other.LiteralCost;
-        this.RedCost = other.RedCost;
-        this.BlueCost = other.BlueCost;
-        this.BitCost = other.BitCost;
-        this.TrivialSymbol = other.TrivialSymbol;
-        this.PaletteCodeBits = other.PaletteCodeBits;
-    }
-
-    /// <summary>
-    /// Initializes a new instance of the <see cref="Vp8LHistogram"/> class.
-    /// </summary>
+    /// <remarks>
+    /// This constructor should be used when the histogram is a member of a <see cref="Vp8LHistogramSet"/>.
+    /// </remarks>
+    /// <param name="buffer">The backing buffer.</param>
     /// <param name="refs">The backward references to initialize the histogram with.</param>
     /// <param name="paletteCodeBits">The palette code bits.</param>
-    public Vp8LHistogram(Vp8LBackwardRefs refs, int paletteCodeBits)
-        : this(paletteCodeBits) => this.StoreRefs(refs);
+    public Vp8LHistogram(Memory<uint> buffer, Vp8LBackwardRefs refs, int paletteCodeBits)
+        : this(buffer, paletteCodeBits) => this.StoreRefs(refs);
 
     /// <summary>
     /// Initializes a new instance of the <see cref="Vp8LHistogram"/> class.
     /// </summary>
+    /// <remarks>
+    /// This constructor should be used when the histogram is a member of a <see cref="Vp8LHistogramSet"/>.
+    /// </remarks>
+    /// <param name="buffer">The backing buffer.</param>
     /// <param name="paletteCodeBits">The palette code bits.</param>
-    public Vp8LHistogram(int paletteCodeBits)
+    /// <param name="bufferOwner">Optional buffer owner to dispose.</param>
+    public Vp8LHistogram(Memory<uint> buffer, int paletteCodeBits, IMemoryOwner<uint>? bufferOwner = null)
     {
+        this.bufferOwner = bufferOwner;
+        this.buffer = buffer;
+        this.bufferHandle = this.buffer.Pin();
```

Comment:
I know I'm the author of the code, but I think this needs further restructuring. The problem is that individual pinning of the buffers forces the original big `IMemoryOwner<uint>` (which is in fact an `UnmanagedBuffer<T> : MemoryManager<T>`) to increase ref counts which will then be only decreased by dispose. This is unnecessary and adds a risk of creating memory leaks. We should instead let `Vp8HistogramSet` do the pinning, so we take an `uint*` here.

For `Vp8HistogramSet` members this would turn `Dispose` into a NOP an `DisposeAt(idx)` an unnecessary method. For better clarity, I would separate the types instead so dispose is only an operation on instances which are actually disposable.

```C#
abstract class Vp8LHistogram
{
    // Hide the ctr
    protected Vp8LHistogram(uint* basePointer);
}
sealed class OwnedVp8LHistogram : Vp8LHistogram, IDisposable
{
    // Static Create methods
}
class Vp8LHistogramSet
{
    // Use the hidden ctr within Vp8LHistogramSet code only.
    private class MemberHistogram : Vp8LHistogram { }
    
    // !!! Delete DisposeAt()
}
```

I can push this change later, after finishing higher priority contributions.

Reply:
This is actually trickier than it looks. Pinning the buffer to pass the pointer is proving troublesome.

Reply:
Figured it out.